bug: ERROR: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1

I can't get start.bat to produce a working demo on Win 11. I get the following errors:

#156 [controls builder 16/16] RUN npm run build
#156 0.861
#156 0.861 > @gigachad-grc/controls@1.0.0 build
#156 0.861 > nest build
#156 0.861
#156 20.25 src/bcdr/bcdr-plans.service.ts:287:13 - error TS2551: Property 'deactivationCriteria' does not exist on type 'UpdateBCDRPlanDto'. Did you mean 'activationCriteria'?
#156 20.25
#156 20.25 287     if (dto.deactivationCriteria !== undefined) {
#156 20.25                 ~~~~~~~~~~~~~~~~~~~~
#156 20.25
#156 20.25   src/bcdr/dto/bcdr.dto.ts:505:3
#156 20.25     505   activationCriteria?: string;
#156 20.25           ~~~~~~~~~~~~~~~~~~
#156 20.25     'activationCriteria' is declared here.
#156 20.25 src/bcdr/bcdr-plans.service.ts:289:23 - error TS2551: Property 'deactivationCriteria' does not exist on type 'UpdateBCDRPlanDto'. Did you mean 'activationCriteria'?
#156 20.25
#156 20.25 289       values.push(dto.deactivationCriteria);
#156 20.25                           ~~~~~~~~~~~~~~~~~~~~
#156 20.25
#156 20.25   src/bcdr/dto/bcdr.dto.ts:505:3
#156 20.25     505   activationCriteria?: string;
#156 20.25           ~~~~~~~~~~~~~~~~~~
#156 20.25     'activationCriteria' is declared here.
#156 20.25 src/bcdr/bcdr-plans.service.ts:292:13 - error TS2339: Property 'objectives' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 292     if (dto.objectives !== undefined) {
#156 20.25                 ~~~~~~~~~~
#156 20.25 src/bcdr/bcdr-plans.service.ts:294:23 - error TS2339: Property 'objectives' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 294       values.push(dto.objectives);
#156 20.25                           ~~~~~~~~~~
#156 20.25 src/bcdr/bcdr-plans.service.ts:297:13 - error TS2339: Property 'assumptions' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 297     if (dto.assumptions !== undefined) {
#156 20.25                 ~~~~~~~~~~~
#156 20.25 src/bcdr/bcdr-plans.service.ts:299:23 - error TS2339: Property 'assumptions' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 299       values.push(dto.assumptions);
#156 20.25                           ~~~~~~~~~~~
#156 20.25 src/bcdr/bcdr-plans.service.ts:302:13 - error TS2339: Property 'planType' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 302     if (dto.planType !== undefined) {
#156 20.25                 ~~~~~~~~
#156 20.25 src/bcdr/bcdr-plans.service.ts:304:23 - error TS2339: Property 'planType' does not exist on type 'UpdateBCDRPlanDto'.
#156 20.25
#156 20.25 304       values.push(dto.planType);
#156 20.25                           ~~~~~~~~
#156 20.25
#156 20.25 Found 8 error(s).
#156 20.25
#156 ERROR: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1

--- a/frontend/src/components/ActionItemsWidget.tsx
+++ b/frontend/src/components/ActionItemsWidget.tsx
@@ -67,6 +67,8 @@ export function ActionItemsWidget({
       const response = await api.get('/api/tasks/my', { params });
       return response.data as Task[];
     },
+    staleTime: 60 * 1000, // 1 minute - prevent excessive refetching
+    refetchOnWindowFocus: false,
   });
 
   // Update task mutation

--- a/frontend/src/components/ActivityFeed.tsx
+++ b/frontend/src/components/ActivityFeed.tsx
@@ -110,6 +110,8 @@ export function ActivityFeed({
       entityType: entityType || (filter !== 'all' ? filter : undefined),
       entityId,
     }).then(res => res.data),
+    staleTime: 60 * 1000, // 1 minute - prevent excessive refetching
+    refetchOnWindowFocus: false,
   });
 
   const activities: AuditLogEntry[] = data?.logs || [];

--- a/frontend/src/components/OnboardingWizard.tsx
+++ b/frontend/src/components/OnboardingWizard.tsx
@@ -37,27 +37,36 @@ export function OnboardingWizard({ onDismiss }: OnboardingWizardProps) {
     }
   }, []);
 
-  // Fetch current state
+  // Fetch current state - use staleTime to prevent refetching on every render
   const { data: frameworks } = useQuery({
     queryKey: ['frameworks'],
     queryFn: () => frameworksApi.list().then((res) => res.data),
+    staleTime: 5 * 60 * 1000, // 5 minutes
+    refetchOnWindowFocus: false,
   });
 
   const { data: controls } = useQuery({
     queryKey: ['controls-count'],
     queryFn: () => controlsApi.list({ limit: 1 }).then((res) => res.data),
+    staleTime: 5 * 60 * 1000, // 5 minutes
+    refetchOnWindowFocus: false,
   });
 
-  const { data: summary } = useQuery({
-    queryKey: ['dashboard-summary'],
-    queryFn: () => dashboardApi.getSummary().then((res) => res.data),
+  // Use the consolidated dashboard-full endpoint to avoid duplicate API calls
+  const { data: dashboardData } = useQuery({
+    queryKey: ['dashboard-full'],
+    queryFn: () => dashboardApi.getFull().then((res) => res.data),
+    staleTime: 5 * 60 * 1000, // 5 minutes
+    refetchOnWindowFocus: false,
   });
+  const summary = dashboardData?.summary;
 
   // Calculate completion status
   const hasFramework = (frameworks?.length || 0) > 0;
   const hasControls = (controls?.meta?.total || 0) > 0;
-  const hasImplementations = (summary?.controlStats?.implemented || 0) > 0;
-  const hasEvidence = (summary?.evidenceStats?.total || 0) > 0;
+  // Handle both old (controlStats) and new (controls) property paths
+  const hasImplementations = (summary?.controls?.byStatus?.implemented || summary?.controlStats?.implemented || 0) > 0;
+  const hasEvidence = (summary?.evidence?.total || summary?.evidenceStats?.total || 0) > 0;
 
   const steps: OnboardingStep[] = [
     {

--- a/services/controls/src/bcdr/dto/bcdr.dto.ts
+++ b/services/controls/src/bcdr/dto/bcdr.dto.ts
@@ -447,6 +447,11 @@ export class UpdateBCDRPlanDto {
   @IsString()
   description?: string;
 
+  @ApiPropertyOptional({ enum: PlanType })
+  @IsOptional()
+  @IsEnum(PlanType)
+  planType?: PlanType;
+
   @ApiPropertyOptional({ enum: PlanStatus })
   @IsOptional()
   @IsEnum(PlanStatus)
@@ -502,7 +507,22 @@ export class UpdateBCDRPlanDto {
   @ApiPropertyOptional()
   @IsOptional()
   @IsString()
+  objectives?: string;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  assumptions?: string;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
   activationCriteria?: string;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  deactivationCriteria?: string;
 
   @ApiPropertyOptional()
   @IsOptional()
@@ -1956,4 +1976,3 @@ export class IncidentFilterDto {
   @IsNumber()
   limit?: number;
 }
-

--- a/services/shared/prisma/schema.prisma
+++ b/services/shared/prisma/schema.prisma
@@ -2816,6 +2816,7 @@ model Vendor {
   @@index([organizationId, category])
   @@index([organizationId, complianceStatus])
   @@index([organizationId, lastReviewedAt])
+  @@index([organizationId, nextReviewDue]) // Added for vendor reviews due dashboard widget
   @@index([workspaceId])
   @@index([createdAt])
   @@map("vendors")

--- a/services/tprm/src/app.module.ts
+++ b/services/tprm/src/app.module.ts
@@ -9,7 +9,7 @@ import { RiskAssessmentModule } from './risk-assessment/risk-assessment.module';
 import { SecurityScannerModule } from './security-scanner/security-scanner.module';
 import { PrismaService } from './common/prisma.service';
 import { AuditService } from './common/audit.service';
-import { StorageModule } from '@gigachad-grc/shared';
+import { StorageModule, CacheModule } from '@gigachad-grc/shared';
 
 @Module({
   imports: [
@@ -17,6 +17,7 @@ import { StorageModule } from '@gigachad-grc/shared';
       isGlobal: true,
     }),
     StorageModule.forRoot(),
+    CacheModule.forRoot({ defaultTtl: 300 }), // 5-minute cache for dashboard widgets
     // RiskAssessmentModule and SecurityScannerModule must be imported BEFORE VendorsModule
     // because their routes (/vendors/:id/risk-assessment/*, /vendors/:id/security-scan/*)
     // are more specific than VendorsModule's catch-all /:id route

--- a/services/tprm/src/vendors/vendors.service.ts
+++ b/services/tprm/src/vendors/vendors.service.ts
@@ -2,6 +2,7 @@ import { Injectable, NotFoundException } from '@nestjs/common';
 import { PrismaService } from '../common/prisma.service';
 import { AuditService } from '../common/audit.service';
 import { TprmConfigService } from '../config/tprm-config.service';
+import { CacheService } from '@gigachad-grc/shared';
 import { CreateVendorDto } from './dto/create-vendor.dto';
 import { UpdateVendorDto } from './dto/update-vendor.dto';
 
@@ -125,6 +126,7 @@ export class VendorsService {
     private readonly prisma: PrismaService,
     private readonly audit: AuditService,
     private readonly tprmConfig: TprmConfigService,
+    private readonly cache: CacheService,
   ) {}
 
   /**
@@ -434,8 +436,22 @@ export class VendorsService {
 
   /**
    * Get vendors due for review with categorization
+   * PERFORMANCE: Cached for 5 minutes to reduce dashboard load times
    */
   async getVendorsDueForReview(organizationId?: string) {
+    const cacheKey = `vendor-reviews-due:${organizationId || 'all'}`;
+    
+    return this.cache.getOrSet(
+      cacheKey,
+      async () => this.getVendorsDueForReviewUncached(organizationId),
+      300, // 5 minute cache
+    );
+  }
+
+  /**
+   * Uncached implementation of getVendorsDueForReview
+   */
+  private async getVendorsDueForReviewUncached(organizationId?: string) {
     const now = new Date();
     const oneWeekFromNow = new Date();
     oneWeekFromNow.setDate(oneWeekFromNow.getDate() + 7);

--- a/services/trust/src/app.module.ts
+++ b/services/trust/src/app.module.ts
@@ -8,12 +8,14 @@ import { TemplatesModule } from './templates/templates.module';
 import { TrustAiModule } from './ai/trust-ai.module';
 import { PrismaService } from './common/prisma.service';
 import { AuditService } from './common/audit.service';
+import { CacheModule } from '@gigachad-grc/shared';
 
 @Module({
   imports: [
     ConfigModule.forRoot({
       isGlobal: true,
     }),
+    CacheModule.forRoot({ defaultTtl: 300 }), // 5-minute cache for dashboard widgets
     QuestionnairesModule,
     KnowledgeBaseModule,
     TrustCenterModule,

--- a/services/trust/src/questionnaires/questionnaires.service.ts
+++ b/services/trust/src/questionnaires/questionnaires.service.ts
@@ -1,6 +1,7 @@
 import { Injectable, NotFoundException } from '@nestjs/common';
 import { PrismaService } from '../common/prisma.service';
 import { AuditService } from '../common/audit.service';
+import { CacheService } from '@gigachad-grc/shared';
 import { CreateQuestionnaireDto } from './dto/create-questionnaire.dto';
 import { UpdateQuestionnaireDto } from './dto/update-questionnaire.dto';
 import { CreateQuestionDto } from './dto/create-question.dto';
@@ -12,6 +13,7 @@ export class QuestionnairesService {
   constructor(
     private prisma: PrismaService,
     private audit: AuditService,
+    private cache: CacheService,
   ) {}
 
   // Questionnaire CRUD
@@ -479,7 +481,21 @@ export class QuestionnairesService {
   }
 
   // Get dashboard data for trust analyst queue widget
+  // PERFORMANCE: Cached for 60s + all queries run in parallel for ~4x faster response
   async getDashboardQueue(organizationId: string, userId?: string) {
+    const cacheKey = `questionnaire-dashboard-queue:${organizationId}`;
+    
+    return this.cache.getOrSet(
+      cacheKey,
+      async () => this.getDashboardQueueUncached(organizationId, userId),
+      60, // 1 minute cache - questionnaire data changes more frequently
+    );
+  }
+
+  /**
+   * Uncached implementation of getDashboardQueue
+   */
+  private async getDashboardQueueUncached(organizationId: string, userId?: string) {
     const now = new Date();
     const oneWeekFromNow = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000);
     const twoWeeksFromNow = new Date(now.getTime() + 14 * 24 * 60 * 60 * 1000);
@@ -490,115 +506,81 @@ export class QuestionnairesService {
       status: { in: [QuestionnaireStatus.pending, QuestionnaireStatus.in_progress] },
     };
 
-    // Get overdue questionnaires
-    const overdue = await this.prisma.questionnaireRequest.findMany({
-      where: {
-        ...baseWhere,
-        dueDate: { lt: now },
+    const selectFields = {
+      id: true,
+      title: true,
+      requesterName: true,
+      company: true,
+      status: true,
+      priority: true,
+      dueDate: true,
+      createdAt: true,
+      _count: { select: { questions: true } },
+      questions: { 
+        where: { status: { in: [QuestionStatus.completed] } },
+        select: { id: true }
       },
-      select: {
-        id: true,
-        title: true,
-        requesterName: true,
-        company: true,
-        status: true,
-        priority: true,
-        dueDate: true,
-        _count: { select: { questions: true } },
-        questions: { 
-          where: { status: { in: [QuestionStatus.completed] } },
-          select: { id: true }
-        },
-      },
-      orderBy: [
-        { priority: 'desc' },
-        { dueDate: 'asc' },
-      ],
-      take: 10,
-    });
+    };
 
-    // Get due this week
-    const dueThisWeek = await this.prisma.questionnaireRequest.findMany({
-      where: {
-        ...baseWhere,
-        dueDate: { gte: now, lte: oneWeekFromNow },
-      },
-      select: {
-        id: true,
-        title: true,
-        requesterName: true,
-        company: true,
-        status: true,
-        priority: true,
-        dueDate: true,
-        _count: { select: { questions: true } },
-        questions: { 
-          where: { status: { in: [QuestionStatus.completed] } },
-          select: { id: true }
+    // Run all queries in parallel for maximum performance
+    const [overdue, dueThisWeek, dueNextWeek, highPriority] = await Promise.all([
+      // Get overdue questionnaires
+      this.prisma.questionnaireRequest.findMany({
+        where: {
+          ...baseWhere,
+          dueDate: { lt: now },
         },
-      },
-      orderBy: [
-        { priority: 'desc' },
-        { dueDate: 'asc' },
-      ],
-      take: 10,
-    });
+        select: selectFields,
+        orderBy: [
+          { priority: 'desc' },
+          { dueDate: 'asc' },
+        ],
+        take: 10,
+      }),
 
-    // Get due next week
-    const dueNextWeek = await this.prisma.questionnaireRequest.findMany({
-      where: {
-        ...baseWhere,
-        dueDate: { gt: oneWeekFromNow, lte: twoWeeksFromNow },
-      },
-      select: {
-        id: true,
-        title: true,
-        requesterName: true,
-        company: true,
-        status: true,
-        priority: true,
-        dueDate: true,
-        _count: { select: { questions: true } },
-        questions: { 
-          where: { status: { in: [QuestionStatus.completed] } },
-          select: { id: true }
+      // Get due this week
+      this.prisma.questionnaireRequest.findMany({
+        where: {
+          ...baseWhere,
+          dueDate: { gte: now, lte: oneWeekFromNow },
         },
-      },
-      orderBy: [
-        { priority: 'desc' },
-        { dueDate: 'asc' },
-      ],
-      take: 5,
-    });
+        select: selectFields,
+        orderBy: [
+          { priority: 'desc' },
+          { dueDate: 'asc' },
+        ],
+        take: 10,
+      }),
 
-    // Get high priority items without due dates
-    const highPriority = await this.prisma.questionnaireRequest.findMany({
-      where: {
-        ...baseWhere,
-        priority: { in: [Priority.high, Priority.critical] },
-        dueDate: null,
-      },
-      select: {
-        id: true,
-        title: true,
-        requesterName: true,
-        company: true,
-        status: true,
-        priority: true,
-        dueDate: true,
-        createdAt: true,
-        _count: { select: { questions: true } },
-        questions: { 
-          where: { status: { in: [QuestionStatus.completed] } },
-          select: { id: true }
+      // Get due next week
+      this.prisma.questionnaireRequest.findMany({
+        where: {
+          ...baseWhere,
+          dueDate: { gt: oneWeekFromNow, lte: twoWeeksFromNow },
         },
-      },
-      orderBy: [
-        { priority: 'desc' },
-        { createdAt: 'asc' },
-      ],
-      take: 5,
-    });
+        select: selectFields,
+        orderBy: [
+          { priority: 'desc' },
+          { dueDate: 'asc' },
+        ],
+        take: 5,
+      }),
+
+      // Get high priority items without due dates
+      this.prisma.questionnaireRequest.findMany({
+        where: {
+          ...baseWhere,
+          priority: { in: [Priority.high, Priority.critical] },
+          dueDate: null,
+        },
+        select: selectFields,
+        orderBy: [
+          { priority: 'desc' },
+          { createdAt: 'asc' },
+        ],
+        take: 5,
+      }),
+    ]);
 
     // Format results with progress calculation
     const formatItem = (item: any) => ({

